annif/templates/home.html

```diff
@@ -29,72 +29,91 @@ <h1 class="py-3">Annif</h1>
   <h2>REST API</h2>
 
   <p>See the <a href="v1/ui/">Swagger documentation</a> for API specification.</p>
-  
+
   <div id="app">
-    <template v-if="errors.length">
+    <template v-if="problems.length">
       <b>Please correct the following error(s):</b>
-      <div class="alert alert-warning" role="alert" v-for="error in errors"><% error %></div>
+      <div :class="['alert', (problem.isError ? 'alert-danger' : 'alert-warning')]"  role="alert" v-for="problem in problems">
+        <h2 v-if="problem.hasTitle()"><% problem.title %></h2>
+        <span v-if="problem.hasDetail()"><% problem.detail %></span>
+      </div>
     </template>
     <div class="row mb-5">
       <div class="col-md-8">
-        <annif-textarea></annif-textarea>
+        <annif-textarea @text-changed="onTextChanged"></annif-textarea>
       </div>
       <div class="col-md-4">
-        <annif-projects></annif-projects>
-        <button type="button" class="btn btn-primary" id="analyze" v-on:click="analyze">Analyze</button>
-        <annif-results></annif-results>
+        <annif-projects :projects="projects" @project-selected="onProjectSelected"></annif-projects>
+        <button type="button" class="btn btn-primary" @click="analyze" :disabled="loading">
+          <span class="spinner-border spinner-border-sm" role="status" aria-hidden="true" v-if="loading"></span>
+          <span v-if="!loading">Analyze</span>
+          <span v-else>Loading...</span>
+        </button>
+        <annif-results :results="results"></annif-results>
       </div>
     </div>
   </div>
 
 </div> <!-- container -->
 
 <script src="/static/js/vue.min.js"></script>
+<!-- bootstrap dependencies -->
 <script src="/static/js/jquery-3.3.1.min.js"></script>
 <script src="/static/js/bootstrap.min.js"></script>
-<script>
+<script type="text/javascript">
 // NB: using delimiters <% %> due to Jinja's syntax taking precedence
 
-// TBD: probably a good idea to use vuex later
-const results_ = [];
+class Problem {
+  constructor(title, detail, isError) {
+    if (title !== undefined) {
+      this.title = title;
+    }
+    if (detail !== undefined) {
+      this.detail = detail;
+    }
+
+    this.isError = !!isError;
+  }
+
+  hasTitle() {
+    return this.title.trim() !== '';
+  }
+
+  hasDetail() {
+    return this.detail.trim() !== '';
+  }
+}
+
+Vue.config.devtools = true;
```

Comment:
This is so that the browser extension can be used to inspect the app. Probably best to remove it when deploying to production. Let me know if you prefer that I remove it, or comment.

Reply:
I've always thought of the Annif UI as a way for the administrator to test projects/models, perhaps also do tasks like setting up projects and training them, but not something you want to expose to the world. So I think enabling the devtools shouldn't do any harm.